// @microsoft/fast-components/dist/esm/data-grid/index.js
import { __decorate } from "tslib";
import { customElement } from "@microsoft/fast-element";
import { DataGrid, createDataGridTemplate, DataGridRow, createDataGridRowTemplate, DataGridCell, createDataGridCellTemplate, } from "@microsoft/fast-foundation";
import { DataGridStyles as gridStyles } from "./data-grid.styles";
import { DataGridRowStyles as rowStyles } from "./data-grid-row.styles";
import { DataGridCellStyles as cellStyles } from "./data-grid-cell.styles";
const gridTemplate = createDataGridTemplate("fast");
const rowTemplate = createDataGridRowTemplate("fast");
const cellTemplate = createDataGridCellTemplate("fast");
/**
 * The FAST Data Grid Element.
 *
 * @public
 * @remarks
 * HTML Element: \<fast-data-grid\>
 */
let FASTDataGrid = class FASTDataGrid extends DataGrid {
};
FASTDataGrid = __decorate([
    customElement({
        name: "fast-data-grid",
        template: gridTemplate,
        styles: gridStyles,
    })
], FASTDataGrid);
export { FASTDataGrid };
/**
 * Styles for DataGrid
 * @public
 */
export const DataGridStyles = gridStyles;
/**
 * The FAST Data Grid Row Element.
 *
 * @public
 * @remarks
 * HTML Element: \<fast-data-grid-row\>
 */
let FASTDataGridRow = class FASTDataGridRow extends DataGridRow {
};
FASTDataGridRow = __decorate([
    customElement({
        name: "fast-data-grid-row",
        template: rowTemplate,
        styles: rowStyles,
    })
], FASTDataGridRow);
export { FASTDataGridRow };
/**
 * Styles for DataGrid row
 * @public
 */
export const DataGridRowStyles = rowStyles;
/**
 * The FAST Data Grid Cell Element.
 *
 * @public
 * @remarks
 * HTML Element: \<fast-data-grid-cell\>
 */
let FASTDataGridCell = class FASTDataGridCell extends DataGridCell {
};
FASTDataGridCell = __decorate([
    customElement({
        name: "fast-data-grid-cell",
        template: cellTemplate,
        styles: cellStyles,
    })
], FASTDataGridCell);
export { FASTDataGridCell };
/**
 * Styles for DataGrid cell
 * @public
 */
export const DataGridCellStyles = cellStyles;
