// @microsoft/fast-components/dist/esm/number-field/index.js
import { __decorate } from "tslib";
import { attr, customElement } from "@microsoft/fast-element";
import { NumberField, NumberFieldTemplate as template } from "@microsoft/fast-foundation";
import { NumberFieldStyles as styles } from "./number-field.styles";
/**
 * The FAST Number Field Custom Element. Implements {@link @microsoft/fast-foundation#NumberField},
 * {@link @microsoft/fast-foundation#NumberFieldTemplate}
 *
 *
 * @public
 * @remarks
 * HTML Element: \<fast-number-field\>
 *
 * {@link https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot/delegatesFocus | delegatesFocus}
 */
let FASTNumberField = class FASTNumberField extends NumberField {
    /**
     * @internal
     */
    connectedCallback() {
        super.connectedCallback();
        if (!this.appearance) {
            this.appearance = "outline";
        }
    }
};
__decorate([
    attr
], FASTNumberField.prototype, "appearance", void 0);
FASTNumberField = __decorate([
    customElement({
        name: "fast-number-field",
        shadowOptions: {
            delegatesFocus: true,
        },
        styles,
        template,
    })
], FASTNumberField);
export { FASTNumberField };
/**
 * Styles for NumberField
 * @public
 */
export const NumberFieldStyles = styles;
